### PR TITLE
feat:[HL-79] Meal Search by Name API 🍕

### PR DESCRIPTION
**Pull request by abdallah-7698**
#### Proposed Changes

Meal Search by Name API

#### Additional Information

Implement the functionality to search for a meal by its name using the API endpoint. Reference URL: [Search meal by name](https://www.themealdb.com/api/json/v1/1/search.php?s=(searchKey))

#### Checklist

- [x] I have tested these changes locally.
- [x] I have added appropriate documentation or updated existing documentation.
- [x] I have added appropriate test coverage or updated existing test coverage.
- [x] I have updated the changelog (if applicable).
- [x] I have followed the project's code style and formatting guidelines.
- [x] I have reviewed and adhered to the project's contributing guidelines.

<!--

### CODE REVIEW

File: Networking/Sources/Networking/Requests/MealSearchRequest.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+// MARK: - SearchMealResponse
+public struct SearchMealResponse: Codable {
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/MealSearchRequest.swift]:
```suggestion
public struct SearchMealResponse: Decodable {
```

File: Networking/Sources/Networking/Requests/MealSearchRequest.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    let meals: [SearchMeal]
+}
+
+// MARK: - SearchMeal
+public struct SearchMeal: Codable {
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/MealSearchRequest.swift]:
```suggestion
public struct SearchMeal: Decodable {
```

**Review by ahmdmhasn — APPROVED** [review]
🚀🚀

**Review by mhmdatallaa — APPROVED** [review]

**Review by esraakhaled — APPROVED** [review]

**Comment by abdallah-7698** [comment]
bbbbb

**Pull request closed without merging**